### PR TITLE
Bump libuv.

### PR DESCRIPTION
**Pull request by maleadt**
Follow-up on https://github.com/JuliaLang/julia/pull/46796; now that all relevant PRs (notably https://github.com/libuv/libuv/pull/3754) have been merged upstream, I've backported this to our branch, https://github.com/JuliaLang/libuv/pull/32, and adapted to the API changes that were involved.

@simonbyrne This should fix the issue you encountered on a SLURM cgroup1 set-up, https://github.com/JuliaLang/julia/pull/46796#issuecomment-1272117544; please verify.

### CODE REVIEW

File: deps/checksums/libuv [hunk 1]
```diff
@@ -1,34 +1,34 @@
...
+LibUV.v2.0.1+13.x86_64-linux-gnu.tar.gz/md5/8c8913068263257cce5042b725918e0e
+LibUV.v2.0.1+13.x86_64-linux-gnu.tar.gz/sha512/a848381012d5a20a0c881f5835e479cfff811928ce508cc57041d69668782f2135c14c7e5388e7dbf693ae57aa1825d911f6f450b9e909cce45487b03a581a23
+LibUV.v2.0.1+13.x86_64-linux-musl.tar.gz/md5/16747c066b6d7fe56850c77f66ea7478
+LibUV.v2.0.1+13.x86_64-linux-musl.tar.gz/sha512/833a02f9191edf3b56f1e02f5671f22de6cb27ec3c9f770530ec95d8da7ba0b9c05bcdf6b094224ea8e43ba70918e1599f3237bd98900763daef80c327d3d2de
+LibUV.v2.0.1+13.x86_64-unknown-freebsd.tar.gz/md5/71f7d9d9234a0623c4b2ee3a44089b62
+LibUV.v2.0.1+13.x86_64-unknown-freebsd.tar.gz/sha512/e73911c3ec35a2201d42c035ecc86e8bd860604b950cb1b7784ff49e27ef5ac9b1da09b59d359ff25b093b87593a8305105bc43711c12eb9654972e280c26d3c
+LibUV.v2.0.1+13.x86_64-w64-mingw32.tar.gz/md5/471d20fa2eac6bfd5d7cdb1b7f58c602
+LibUV.v2.0.1+13.x86_64-w64-mingw32.tar.gz/sha512/3f5ad55268184227378ddcfed0146bf0386c8cf468bc53a348d21195d818db4db768be61fd23e1ee2ecbb52f073815884a04a923d815b9b5992825d144c0633a
 libuv-e6f0e4900e195c8352f821abe2b3cffc3089547b.tar.gz/md5/c4465d7bff6610761cf37a1e8e3da08c
```
Comment (giordano) [inline deps/checksums/libuv:33]: I guess you missed to update https://github.com/JuliaLang/julia/blob/04214ece63e472fada63f2ff961b5d10d925669a/deps/libuv.version#L7

**Comment by maleadt** [comment]
@gbaraldi This also includes https://github.com/JuliaLang/libuv/pull/29, which AFAIK you hadn't bumped to master yet?

**Comment by simonbyrne** [comment]
It works! Many thanks @maleadt for seeing this through.

```
┌─[8]──[Tue Nov 29]──[17:08:08]────────────────────────────────────────
│ spjbyrne@login2:~/julia-6ba9fd56e5
├ srun --reservation=clima -t 00:20:00 --mem 4G --pty bash -l

┌─[1]──[Tue Nov 29]──[17:08:18]────────────────────────────────────────
│ spjbyrne@hpc-91-17:~/julia-6ba9fd56e5
├ bin/julia
               _
   _       _ _(_)_     |  Documentation: https://docs.julialang.org
  (_)     | (_) (_)    |
   _ _   _| |_  __ _   |  Type "?" for help, "]?" for Pkg help.
  | | | | | | |/ _` |  |
  | | |_| | | | (_| |  |  Version 1.10.0-DEV.75 (2022-11-30)
 _/ |\__'_|_|_|\__'_|  |  tb/libuv/6ba9fd56e5c (fork: 3 commits, 0 days)
|__/                   |

julia> Sys.total_memory() / 2^20
4096.0

julia> Sys.total_physical_memory() / 2^20
191858.4375
```

**Review by simonbyrne — APPROVED** [review]

**Comment by maleadt** [comment]
Thanks for verifying. CI failures look unrelated, so this seems good to go.

**Comment by DilumAluthge** [comment]
Looks like this PR did not update the source checksums for libuv.

**Pull request merged**